### PR TITLE
Support login to torec, if no login or login has failed - failing back to guest user

### PR DESCRIPTION
**Pull request by gabik**
Hi,

- Fixed the login implementation
- Trying to login if user filled credentials on settings
- if login succeeded - trying to download with authorized user
- if login failed (or credentials not filled) continuing your regular guest flow

Please review.

Thanks,
Gabi.

### CODE REVIEW

**Comment by gabik** [comment]
Good catch. Fixed.

**Comment by gabik** [comment]
There is small issue with the cookie I am fixing. please wait with the merge.

**Comment by gabik** [comment]
Please check 1.4.1 instead of 1.4
Fixed cookie jar permissions on save.

File: resources/lib/TorecSubtitlesDownloader.py [hunk 1]
```diff
@@ -306,6 +327,27 @@ def _get_download_link_with_regular_token(self, sub_id, option_id):
     def get_download_link(self, sub_id, option_id):
         self._confirm_download_code(sub_id, option_id)
 
+        # Checking if we have cookie - if so, we will try to use it
+        log(__name__, "Checking if we have cookie")
```
Comment (slmosl) [inline resources/lib/TorecSubtitlesDownloader.py]: extract all of this into a new `tryLogin` function

File: resources/lib/TorecSubtitlesDownloader.py [hunk 2]
```diff
@@ -267,13 +285,16 @@ def _confirm_download_code(self, sub_id, option_id):
         response      = self.opener.open("%s/ajax/sub/t7/guest_dl_code.asp" % self.BASE_URL, urllib.urlencode(params))
         response_data = response.read()
 
-    def _try_get_download_link(self, sub_id, option_id, guest_token):
+    def _try_get_download_link(self, sub_id, option_id, guest_token, is_login=False):
+        if guest_token is None:
```
Comment (slmosl) [inline resources/lib/TorecSubtitlesDownloader.py]: why are you doing this?

**Comment by gabik** [comment]
@slmosl - 1.4.2 Changes:
- Added to `changelog.txt`
- separate to `_get_download_link_with_login` function (Called before guest tries)

About the tests - the download flow is the same as before, the only thing you should do in order to test login donload is to add your creds into `xbmcaddon.py` file, see method `getSetting`

Do you see any other changes we need on this request?

About tests:
tested on Android, RPi (OpenElec), Windows. both Kodi 16 and 17.

File: resources/lib/TorecSubtitlesDownloader.py (hunk 2 above)
Comment (gabik) [inline resources/lib/TorecSubtitlesDownloader.py]: To be able to get `guest_token` both as empty string and None, if getting None I am converting to empty string. I need it to be empty string to send to Torec.

File: addon.xml [hunk 3]
```diff
@@ -1,7 +1,7 @@
 <?xml version="1.0" encoding="UTF-8" standalone="yes"?>
 <addon id="service.subtitles.torec"
        name="Torec"
-       version="1.3.1"
+       version="1.4.2"
```
Comment (slmosl) [inline addon.xml]: there's no point in bumping up version, until published - so change to `1.4.0` here

File: changelog.txt [hunk 4]
```diff
@@ -1,3 +1,15 @@
+1.4.2
```
Comment (slmosl) [inline changelog.txt]: coalesce all the `1.4.*` changes to `1.4.0`

**Comment by gabik** [comment]
@slmosl - Done. Sorry for the little mess, each repository and its own rules 👍 
Will know from now on.. 🥇

**Pull request merged**